mailbox/plugin/quota-mailing/src/main/java/org/apache/james/eventsourcing/CommandDispatcher.java

```diff
@@ -0,0 +1,77 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.eventsourcing;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.IntStream;
+
+import com.google.common.collect.ImmutableList;
+
+public class CommandDispatcher {
+
+    private static final int MAX_RETRY = 10;
+
+    public interface Command {
+    }
+
+    public interface CommandHandler<C extends Command> {
+        Class<C> getCommandClass();
+
+        List<? extends Event> handle(C c);
+    }
+
+    private final EventBus eventBus;
+    private final Map<Class, CommandHandler> handlers;
+
+    public CommandDispatcher(EventBus eventBus) {
+        this.eventBus = eventBus;
+        this.handlers = new ConcurrentHashMap<>();
```

Comment:
not sure we should use that, we don't know what happens on dynamic changes so we could probably just have a  Dispatcher builder that create an Immutable Dispatcher with an ImmutableMap, WDYT ?

Reply:
Would seems to be a great idea.

Anyway, such a builder might not even be needed: all the commandHandlers are known in the event system...